code/modules/atmospherics/environmental/LINDA_system.dm

```diff
@@ -49,18 +49,9 @@
 	var/canpass = CANATMOSPASS(src, src)
 	var/canvpass = CANVERTICALATMOSPASS(src, src)
 	// I am essentially inlineing two get_dir_multizs here, because they're way too slow on their own. I'm sorry brother
-	var/list/z_traits = SSmapping.multiz_levels[z]
 	for(var/direction in GLOB.cardinals_multiz)
-		// Yes this is a reimplementation of get_step_mutliz. It's faster tho. fuck you
-		var/turf/current_turf = (direction & (UP|DOWN)) ? \
-			(direction & UP) ? \
-				(z_traits[Z_LEVEL_UP]) ? \
-					(get_step(locate(x, y, z + 1), NONE)) : \
-				(null) : \
-				(z_traits[Z_LEVEL_DOWN]) ? \
-					(get_step(locate(x, y, z - 1), NONE)) : \
-				(null) : \
-			(get_step(src, direction))
+		// Get step multi-z is cached now so is much faster
+		var/turf/current_turf = get_step_multiz(src, direction)
```

Comment:
It's a lot more challenging since the function holds a local state, however this function doesn't seem to work with non-cardinal directions so if that assumption holds then its probably technically possible

Reply:
I have inlined the entire proc, meaning that all usecases will get the advantage of being inlined